src/main/webapp/WEB-INF/view/map.jsp

```diff
@@ -56,30 +66,49 @@
 
   // This event listener calls addMarker() when the map is clicked.
         google.maps.event.addListener(map, 'click', function(event) {
-          lat = event.latLng.lat();
-          lng = event.latLng.lng();
           addMarker(event.latLng, map);
+          document.querySelector("#latitudeVal").value = event.latLng.lat();
+          document.querySelector("#longitudeVal").value = event.latLng.lng();
         });
+
+        for (var i in markerLocs) {
+           addMarkerFromStore(markerLocs[i], map, locNames[i], convoNames[i])
+        }
   }
 
+  // Adds a markerstore marker to the map.
+      function addMarkerFromStore(location, map, markerName, convoName) {
+        var contentString = '<h3 align="center">' + markerName +'</h3>' + '<a href="/chat/' + convoName 
+        +'">Click here</a> to join the conversation going on at this location!';
```

Comment:
Minor nit: Consider removing the "Click here" and make "Join" or "Join the conversation" the link text instead. The idea is put the focus on the desired action instead of the clicking.